be1-go/message/answer/error.go

```diff
@@ -44,3 +44,43 @@ func NewInvalidObjectError(action string) *Error {
 		Description: fmt.Sprintf("invalid object: %s", action),
 	}
 }
+
+// NewInvalidResourceError returns an error with -2 for an object with invalid ressources
```

Comment:
You should have a generic error function, and then specialized error functions that use it. You can also make those functions accept a format and arbitrary args, in go we then end those functions with `f`: 

```go
func NewError(code int, desc string) {
	return &Error{
		Code:       code,
		Description: desc,
	}
}

func NewErrorf(code int, format string, a ...interface{}) {
  return NewError(code, fmt.Sprintf(format, ...a))
}

func NewInvalidResourceErrorf(format string, a ...interface{}) {
  return NewErrorf(-2, format, ...a)
}

...
```

Reply:
Also please format comments at 80 chars length.